server/controllers/moleculesImporter.js

```diff
@@ -0,0 +1,262 @@
+import fs from "fs/promises";
+import path from "path";
+
+import { queryPromise } from "../db/database.js";
+import { HeaderErrors } from "../global/csv_reader/HeaderChecker.js";
+import HttpResponseWrapper from "../global/HttpResponseWrapper.js";
+import Logger, { addErrorTitle } from "../global/Logger.js";
+import { analyzeData } from "../global/molecules_analyzer/moleculesAnalyzer.js";
+import { createSqlToInsertAllData } from "../global/molecules_importer/moleculesImporter.js";
+import { parseMoleculesFromCsv } from "../global/molecules_parser/Parser.js";
+
+/**
+ *
+ * @api {post} /import/molecules Import a csv file of molecules data
+ * @apiName ImportMolecules
+ * @apiGroup Import
+ * 
+ * @apiPermission LoggedIn
+ * @apiPermission Admin
+ * @apiDescription Import a csv file to update the molecules data, the format of the request must be multipart/form-data ! 
+ * 
+ * @apiParam  {File} file The csv file
+ * @apiParam {string} confirmed If "true", the data will be imported, otherwise they will be only tested
+ * 
+ * @apiSuccess (200) {string} message Message explaining what was done
```

Comment:
May we have a different response code to differenciate a correct file and an imported file ?
[201 Created and 202 Accepted](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses) could be good candidates IMHO.

Reply:
Yep good idea !